pkg/placement/apiwatch-map-provider.go

```diff
@@ -21,38 +21,35 @@ import (
 	"fmt"
 	"sync"
 
-	apiextinformers "k8s.io/apiextensions-apiserver/pkg/client/kcp/informers/externalversions/apiextensions/v1"
+	apiextclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextinfactory "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
 	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/discovery"
 	upstreamcache "k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog/v2"
 
-	clusterdiscovery "github.com/kcp-dev/client-go/discovery"
-	bindinginformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/apis/v1alpha1"
-	"github.com/kcp-dev/logicalcluster/v3"
-
 	ksmetav1a1 "github.com/kubestellar/kubestellar/pkg/apis/meta/v1alpha1"
 	apiwatch "github.com/kubestellar/kubestellar/pkg/apiwatch"
+	msclient "github.com/kubestellar/kubestellar/space-framework/pkg/msclientlib"
 )
 
 // NewAPIWatchMapProvider constructs an APIMapProvider that gets its information
 // from apiwatch.
 func NewAPIWatchMapProvider(ctx context.Context,
 	numThreads int,
-	discoveryClusterClient clusterdiscovery.DiscoveryClusterInterface,
-	crdClusterPreInformer apiextinformers.CustomResourceDefinitionClusterInformer,
-	bindingClusterPreInformer bindinginformers.APIBindingClusterInformer,
+	spaceclient msclient.KubestellarSpaceInterface,
+	spaceProviderNs string,
 ) APIWatchMapProvider {
 	awp := &apiWatchProvider{
-		context:                   ctx,
-		numThreads:                numThreads,
-		discoveryClusterClient:    discoveryClusterClient,
-		crdClusterPreInformer:     crdClusterPreInformer,
-		bindingClusterPreInformer: bindingClusterPreInformer,
-		queue:                     workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
-		perCluster:                NewMapMap[logicalcluster.Name, *apiWatchProviderPerCluster](nil),
+		context:         ctx,
+		numThreads:      numThreads,
+		queue:           workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		perCluster:      NewMapMap[string, *apiWatchProviderPerCluster](nil),
+		spaceclient:     spaceclient,
```

Comment:
This is not a "client" of a "space". This is something else. So I would name this field something more aligned with the functionality involved.